### PR TITLE
Migrate to spring 5

### LINKED ISSUE #839: Migrate to spring5

### PR DESCRIPTION
**Pull request by jimirocks**
Resolves #839

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by liry** [comment]
Does this force user to use newest Spring?
FYI: Spring 4.3.x comes with extended maintenance until at least June 2020

**Comment by jimirocks** [comment]
@liry as far as I could see, the user is not forced to spring5

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

File: gooddata-java/src/main/java/com/gooddata/sdk/service/project/ProjectService.java [hunk 1]
```diff
@@ -64,7 +65,7 @@
  * Usage example:
  * <pre><code>
  *     ProjectService projectService = gd.getProjectService();
- *     Collection&lt;Project&gt; projects = projectService.getProjects();
+ *     Collection&lt;Project&gt; projects = projectService.getProjects();A
```
Comment (peter-plochan) [inline gooddata-java/src/main/java/com/gooddata/sdk/service/project/ProjectService.java:68]:
```suggestion
 *     Collection&lt;Project&gt; projects = projectService.getProjects();
```

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-java-merge-compliance-check/137/ : SUCCESS in 11s

**Pull request merged**